### ISSUE TITLE
Research: come up with 3-4 algos/models

### ISSUE THREAD

**Issue by shannonhlo** [shannonhlo/pokemon_classification#10]
Come up with 3-4 algos/models to try for binary classification

**Comment by nkumar97** [comment]
Tree Based Models: Divides data up into regions based on "if-then" type of questions 
Pro: More flexible
Con: So flexible they can lead to overfitting

Models to use:

For high interpretability: classification and regression trees (CART)
For high performance: xgboost, random forest --> these are known as ensemble models, they combine multiple learning algorithms together to achieve peak results but tend to be a black box

Linear Models: Linear models attempt to take the feature space and draw a straight line somewhere that creates an accurate division of the two classes of the response variable 

Models to use:

Logistic regression: simple with high interpretability but likely lower performing than any tree based model

**Comment by shannonhlo** [comment]
Naive Bayes
- assumes each input variables is indepedent
- gives probability of hypothesis happening given evidence

Support Vector Machine
- algorithm creates a line or a hyperplane which separates the data into classes
- https://towardsdatascience.com/https-medium-com-pupalerushikesh-svm-f4b42800e989

Random Forest
- A large number of relatively uncorrelated models (trees) operating as a committee will outperform any of the individual constituent models. ([link](https://towardsdatascience.com/understanding-random-forest-58381e0602d2#:~:text=a%20large%20number%20of%20relatively%20uncorrelated%20models%20(trees)%20operating%20as%20a%20committee%20will%20outperform%20any%20of%20the%20individual%20constituent%20models.))

Convolutional Neural Network
- https://towardsdatascience.com/a-comprehensive-guide-to-convolutional-neural-networks-the-eli5-way-3bd2b1164a53
